Source/JavaScriptCore/jsc.cpp

```diff
@@ -3179,7 +3179,7 @@ JSC_DEFINE_HOST_FUNCTION(functionGenerateHeapSnapshot, (JSGlobalObject* globalOb
     }
 
     EncodedJSValue result = JSValue::encode(JSONParse(globalObject, jsonString));
-    scope.releaseAssertNoException();
+    RETURN_IF_EXCEPTION(scope, { });
```

Comment:
Maybe we should use `RELEASE_AND_RETURN` here?

Reply:
Yeah, we should just do `RELEASE_AND_RETURN(scope, JSValue::encode(JSONParse(globalObject, jsonString)));` since `JSONParse` can throw exceptions. And we should also fix the other `JSONParse` uses in our code base.